dali/pipeline/executor/executor.cc

```diff
@@ -356,10 +356,16 @@ int Executor<WorkspacePolicy, QueuePolicy>::InferBatchSize(
     DALI_ENFORCE(bsps[0]->NextBatchSize() == bsps[i]->NextBatchSize(),
                  "Batch size must be uniform across an iteration");
   }
-  auto batch_size = bsps[0]->NextBatchSize();
-  assert(batch_size > 0);
-  for (auto &bsp : bsps) {
-    bsp->Advance();
+  try {
+    auto batch_size = bsps[0]->NextBatchSize();
+    assert(batch_size > 0);
+    for (auto &bsp : bsps) {
+      bsp->Advance();
+    }
+  } catch (const std::exception &e) {
+    DALI_FAIL(
+        make_string("Failed to acquire the next batch. Make sure, that DALI Pipeline is fed "
+                    "with sufficient amount of data. ", e.what()));
```

Comment:
Can we have a name of the offending node here? If the node was populated with `feed_input`, we'll have the name of the very source the user failed to feed.

Reply:
Usually the name is generated and doesn't tell much.

Reply:
Not for external_source that's manually fed with `feed_input`, usually _by name_ (like in Triton) - and for others it would at least be "ExternalSource_<index>" and that index would most likely correspond to the ordinal of the external source in order of operator instantiation.